File: app/src/main/java/com/examle/curexchange/ui/base/BaseActivity.java
package com.examle.curexchange.ui.base;

import android.app.ProgressDialog;
import android.widget.ProgressBar;
import android.widget.Toast;

import com.arellomobile.mvp.MvpAppCompatActivity;

public class BaseActivity extends MvpAppCompatActivity implements BaseView {

    private ProgressDialog progressDialog;

    @Override
    public void showToast(String error) {
        Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void showProgress(boolean inProgress) {
        if (inProgress) {
            if (progressDialog != null) {
                progressDialog.show();
            } else {
                progressDialog = ProgressDialog.show(this, null, "Please wait while download will be success", true, false);
                progressDialog.show();
            }
        } else {
            if (progressDialog != null) progressDialog.dismiss();
        }
    }
}

File: app/src/main/java/com/examle/curexchange/ui/base/BaseActivity.kt
package com.examle.curexchange.ui.base

import android.app.ProgressDialog
import android.widget.Toast
import com.arellomobile.mvp.MvpAppCompatActivity

open class BaseActivity : MvpAppCompatActivity(), BaseView {

    var progressDialog: ProgressDialog? = null

    override fun showToast(error: String) {
        Toast.makeText(this, error, Toast.LENGTH_SHORT).show()
    }

    override fun showProgress(isShow: Boolean) {
        if (isShow) {
            if (progressDialog != null) {
                progressDialog!!.show()
            } else {
                progressDialog = ProgressDialog.show(this, null, "Please wait while download will be success", true, false)
                progressDialog!!.show()
            }
        } else {
            if (progressDialog != null) progressDialog!!.dismiss()
        }
    }
}